src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,258 @@
+        }
+        edge_.set_dst(Expression::asInt(dst.value()));
+        edge_.set_ranking(sentence_->getRank());
+
+        auto spaceId = ectx()->rctx()->session()->space();
+        edgeTypeName_ = sentence_->getEdgeType();
+        auto edgeStatus = ectx()->schemaManager()->toEdgeType(spaceId, *edgeTypeName_);
+        if (!edgeStatus.ok()) {
+            status = edgeStatus.status();
+            break;
+        }
+        auto edgeType = edgeStatus.value();
+        edge_.set_edge_type(edgeType);
+
+        status = prepareSet();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareWhen();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareYield();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    return status;
+}
+
+
+Status UpdateEdgeExecutor::prepareSet() {
+    auto status = Status::OK();
+    auto items = sentence_->updateList()->items();
+    for (auto& item : items) {
+        // item: name_label ASSIGN expression
+        storage::cpp2::UpdateItem updateItem;
+        updateItem.name = *edgeTypeName_;
+        auto propName = item->field();
+        updateItem.prop = *propName;
+        updateItem.value = Expression::encode(item->value());
+        updateItems_.emplace_back(std::move(updateItem));
+    }
+    if (updateItems_.empty()) {
+        status = Status::Error("There must be some correct update items.");
+    }
+    return status;
+}
+
+
+Status UpdateEdgeExecutor::prepareWhen() {
+    auto *clause = sentence_->whenClause();
+    if (clause != nullptr) {
+        filter_ = clause->filter();
+    }
+    return Status::OK();
+}
+
+
+Status UpdateEdgeExecutor::prepareYield() {
+    auto *clause = sentence_->yieldClause();
+    if (clause != nullptr) {
+        yields_ = clause->columns();
+    }
+    return Status::OK();
+}
+
+
+std::vector<std::string> UpdateEdgeExecutor::getReturnColumns() {
+    std::vector<std::string> returnColumns;
+    for (auto *col : yields_) {
+        auto column = Expression::encode(col->expr());
+        returnColumns.emplace_back(std::move(column));
+    }
+    return returnColumns;
+}
+
+
+void UpdateEdgeExecutor::finishExecution(storage::cpp2::UpdateResponse &&rpcResp) {
+    resp_ = std::make_unique<cpp2::ExecutionResponse>();
+    std::vector<std::string> columnNames;
+    columnNames.reserve(yields_.size());
+    for (auto *col : yields_) {
+        if (col->alias() == nullptr) {
+            columnNames.emplace_back(col->expr()->toString());
+        } else {
+            columnNames.emplace_back(*col->alias());
+        }
+    }
+    resp_->set_column_names(std::move(columnNames));
+
+    std::vector<cpp2::RowValue> rows;
+    if (rpcResp.__isset.schema && rpcResp.__isset.data) {
+        std::shared_ptr<ResultSchemaProvider> schema;
+        if (rpcResp.get_schema() != nullptr) {
+            schema = std::make_shared<ResultSchemaProvider>(rpcResp.schema);
+        }
```

Comment:
what about rpcResp.get_schema() == nullptr?

Reply:
done, no check it.